mix.exs

```diff
@@ -1,29 +1,45 @@
 defmodule TeslaApi.MixProject do
   use Mix.Project
 
+  @version "0.1.0"
+  @url "https://github.com/mgwidmann/tesla_api"
+
   def project do
     [
       app: :tesla_api,
-      version: "0.1.0",
+      version: @version,
       elixir: "~> 1.8",
       start_permanent: Mix.env() == :prod,
-      deps: deps()
+      deps: deps(),
+      aliases: [docs: &build_docs/1]
     ]
   end
 
   # Run "mix help compile.app" to learn about applications.
   def application do
-    [
-      extra_applications: [:logger]
-    ]
+    []
   end
 
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:tesla, "~> 1.2.0"},
-      {:hackney, "~> 1.14.0"},
-      {:jason, ">= 1.0.0"}
+      {:tesla, "~> 1.2"},
+      {:hackney, "~> 1.14"},
+      {:jason, "~> 1.0"}
     ]
   end
+
+  defp build_docs(_) do
+    Mix.Task.run("compile")
+    ex_doc = Path.join(Mix.Local.path_for(:escript), "ex_doc")
```

Comment:
This works I guess but isn't it easier to just add ex_doc as a doc only dependency? Why do all this?

Reply:
[nimble_parsec](https://github.com/plataformatec/nimble_parsec/blob/master/mix.exs) does it like that. But I really don't have a strong opinion here. I will remove it for now. You should decide when releasing to hex.pm.